apps/api/src/account/account.controller.ts

```diff
@@ -1,29 +1,39 @@
-import { Controller, Get } from '@nestjs/common'
+import { Body, Controller, Get, Put } from '@nestjs/common'
 import { RealmViewSupporters, ViewSupporters } from '@podkrepi-bg/podkrepi-types'
 import { AuthenticatedUser, Public, RoleMatchingMode, Roles } from 'nest-keycloak-connect'
 
 import { KeycloakTokenParsed } from '../auth/keycloak'
+import { UpdatePersonDto } from '../person/dto/update-person.dto'
+import { PersonService } from '../person/person.service'
 import { AccountService } from './account.service'
 
 @Controller('account')
 export class AccountController {
-  constructor(private readonly accountService: AccountService) {}
+  constructor(private readonly accountService: AccountService, private readonly personService: PersonService) {}
 
   @Get('me')
   @Public(false)
-  getMe(
+  async getMe(
     @AuthenticatedUser()
     user: KeycloakTokenParsed,
   ) {
     if (user) {
       // Public authenticated
       console.log(user)
-      return { status: 'authenticated', user }
+      return await this.personService.findOne(user.sub as string);
     }
     // Public
     return { status: 'unauthenticated' }
   }
 
+  @Put('me')
+  async updateProfile(@AuthenticatedUser() user: KeycloakTokenParsed, @Body() data: UpdatePersonDto) {
+    if (data.birthday) {
+      data.birthday = new Date(data.birthday);
```

Comment:
We may try to wrap this in a try-catch block as if a bad date is provided in `birthday` field it would result in error 500